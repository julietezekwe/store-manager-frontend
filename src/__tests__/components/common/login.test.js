/* eslint-disable react/react-in-jsx-scope */
/* eslint-disable no-undef */
import { Login } from '../../../components/common/Login.jsx';

let wrapper;

describe('<LoginPage />', () => {
  beforeEach(() => {
    const props = {
      loginUser: () => Promise.resolve(),
      login: {
        isLoggedIn: false,
        user: null,
      },
    };
    wrapper = shallow(<Login {...props} />);
  });

  it('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('calls onButtonSubmit()', () => {
    const values = {
      password: 'admin',
      email: 'admin@gmail.com',
    };
    sinon.spy(wrapper.instance(), 'onButtonSubmit');
    wrapper.instance().onButtonSubmit(values);
    expect(wrapper.instance().onButtonSubmit.calledOnce)
      .toEqual(true);
    expect(wrapper.instance().onButtonSubmit.calledWith(values));
  });
  it('calls onChangeHandler()', () => {
    const event = {
      target: {
        id: 'email',
        value: 'admin@gmail.com',
      },
    };
    sinon.spy(wrapper.instance(), 'onChangeHandler');
    wrapper.instance().onChangeHandler(event);
    expect(wrapper.instance().onChangeHandler.calledOnce)
      .toEqual(true);
    expect(wrapper.instance().onChangeHandler.calledWith(event));
  });
});
