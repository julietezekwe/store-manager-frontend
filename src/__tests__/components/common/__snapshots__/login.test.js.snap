// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoginPage /> renders correctly 1`] = `
<section
  className="login-page"
>
  <div
    className="login-form"
  >
    <h1
      className="login-heading"
    >
      Log In
    </h1>
    <div
      className="myCard col-9"
      id="alert"
    />
    <div
      className="form login"
    >
      <div>
        <input
          id="email"
          onChange={[Function]}
          placeholder="Email"
          required={true}
          type="email"
        />
      </div>
      <div>
        <input
          id="password"
          onChange={[Function]}
          placeholder="password"
          required={true}
          type="password"
        />
      </div>
      <div>
        <input
          onClick={[Function]}
          type="submit"
          value="LOG IN"
        />
      </div>
      <div
        className="forgotpassword"
      >
        <p>
          <i>
            Forgot Password?
          </i>
        </p>
      </div>
    </div>
  </div>
</section>
`;
